js/mcqView.js

```diff
@@ -1,175 +1,86 @@
-], function(Adapt, QuestionView) {
-
-  var McqView = QuestionView.extend({
-
-    events: {
-      'focus .js-item-input': 'onItemFocus',
-      'blur .js-item-input': 'onItemBlur',
-      'change .js-item-input': 'onItemSelect',
-      'keyup .js-item-input': 'onKeyPress'
-    },
-
-     isCorrectAnswerShown: false,
-
-    initialize: function() {
-      QuestionView.prototype.initialize.apply(this, arguments);
-      this.update = _.debounce(this.update.bind(this), 1);
-      this.listenTo(this.model, {
-        "change:_isEnabled change:_isComplete change:_isSubmitted": this.update
-      });
-      this.listenTo(this.model.getChildren(), {
-        "change:_isActive": this.update
-      });
-    },
-
-    resetQuestionOnRevisit: function() {
-      this.resetQuestion();
-    },
-
-    setupQuestion: function() {
-      this.model.setupRandomisation();
-    },
-
-    onQuestionRendered: function() {
-      this.setReadyStatus();
-      this.update();
-    },
-
-    onKeyPress: function(event) {
-      if (event.which !== 13) return;
-      //<ENTER> keypress
-      this.onItemSelect(event);
-    },
-
-    onItemFocus: function(event) {
-      if (!this.model.isInteractive()) return;
-
-      this.$('.js-item-label[for='+$(event.currentTarget).attr('id')+']').addClass('is-highlighted');
-    },
-
-    onItemBlur: function(event) {
-      this.$('.js-item-label[for='+$(event.currentTarget).attr('id')+']').removeClass('is-highlighted');
-    },
-
-    onItemSelect: function(event) {
-      if (!this.model.isInteractive()) return;
-
-      var index = $(event.currentTarget).data('adapt-index');
-      var itemModel = this.model.getItem(index);
-      var shouldSelect = !itemModel.get("_isActive");
-
-      if (this.model.isSingleSelect()) {
-        // Assume a click is always a selection
-        shouldSelect = true;
-        this.model.resetActiveItems();
-      } else if (shouldSelect && this.model.isAtActiveLimit()) {
-        // At the selection limit, deselect the last item
-        this.model.getLastActiveItem().toggleActive(false);
-      }
-
-      // Select or deselect accordingly
-      itemModel.toggleActive(shouldSelect);
-    },
-
-    // Blank method to add functionality for when the user cannot submit
-    // Could be used for a popup or explanation dialog/hint
-    onCannotSubmit: function() {},
-
-    // This is important and should give the user feedback on how they answered the question
-    // Normally done through ticks and crosses by adding classes
-    showMarking: function() {
-      this.update();
-    },
-
-    // Used by the question view to reset the look and feel of the component.
-    resetQuestion: function() {
+import QuestionView from 'core/js/views/questionView';
+
+class McqView extends QuestionView {
+
+  initialize(...args) {
+    super.initialize(...args);
+    this.listenTo(this.model, {
+      "_isEnabled _isComplete _isSubmitted _isCorrectAnswerShown": this.changed
+    });
```

Comment:
just to check, is it definitely OK to leave out the 'change:' prefix?

Reply:
This whole section should be unnecessary. The QuestionView inherits AdaptView which handles the change events from the model automatically. It's only necessary to plumb in the getChildren collection as there is no ItemsQuestionView to handle that behaviour automatically for you.